### PR TITLE
Added German slugifying plugin.

### PR DESCRIPTION
**Pull request by felixfontein**
Uses Nikola's newest feature (https://github.com/getnikola/nikola/pull/2284) to transform ä,ö,ü,ß to ae,oe,ue,ss for German languages.

### CODE REVIEW

**Comment by felixfontein** [comment]
Fixes getnikola/nikola#2284 and getnikola/nikola#2273.

File: v7/german_slugify/german_slugify.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+from nikola.plugin_categories import SignalHandler
+from nikola import utils
+
+
+def _needs_german_slugifying_rules(lang, locale):
+    """Checks whether the given language name with corresponding locale needs German slugifying rules."""
+    if locale.split('_')[0] in {'de', 'gsw'}:  # German and Allemanic German
+        return True
+    if lang in {'de', 'deutsch', 'german',
```
Comment (Kwpolska) [inline v7/german_slugify/german_slugify.py]: Where would you get all those broken strings from?

File: v7/german_slugify/german_slugify.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    def set_site(self, site):
+        super(GermanSlugify, self).set_site(site)
+
+        # First determine languages which belong to German locales
+        self.german_languages = set()
+        for lang, loc in utils.LocaleBorg.locales.items():
+            if _needs_german_slugifying_rules(lang, loc):
+                self.german_languages.add(lang)
+        if len(self.german_languages) > 0:
+            utils.LOGGER.info("Using German slugifying rules for the following languages: {0}".format(', '.join(sorted(self.german_languages))))
```
Comment (Kwpolska) [inline v7/german_slugify/german_slugify.py]: `.debug()`

File: v7/german_slugify/german_slugify.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        # Store old slugify method
+        self.old_slugify = utils.slugify
+
+        # Define new slugify method
+        def new_slugify(value, lang=None, force=False):
+            if lang is not None and lang in self.german_languages:
+                if not isinstance(value, utils.unicode_str):
+                    raise ValueError("Not a unicode object: {0}".format(value))
+                value = value.replace('ä', 'ae').replace('ö', 'oe').replace('ü', 'ue').replace('ß', 'ss')
```
Comment (Kwpolska) [inline v7/german_slugify/german_slugify.py]: ß is handled by `old_slugify` properly.

Comment (felixfontein) [inline v7/german_slugify/german_slugify.py]: I know, I only added it for sake of completeness.

File: v7/german_slugify/german_slugify.py (hunk 2 above)
Comment (felixfontein) [inline v7/german_slugify/german_slugify.py]: Good point. Not everyone likes my plugin as noisy as me ;-)

File: v7/german_slugify/german_slugify.py (hunk 1 above)
Comment (felixfontein) [inline v7/german_slugify/german_slugify.py]: I removed `'deutsch'` and `'german'`. I left the others as I don't know of another way to distinguish between different german locales.

**Comment by felixfontein** [comment]
Thanks for merging!

**Pull request merged**